File: frontend/src/services/aiService.js
import api from './api';

export const aiService = {
  // Chat with AI Tutor
  async chatWithAI(message, context = {}) {
    try {
      const res = await api.post('/ai/chat', { message, context });
      return res.data;
    } catch (error) {
      console.warn('Backend offline, using intelligent client simulation for chat', error);
      // Simulate tutor thinking delay
      await new Promise(r => setTimeout(r, 1200));

      const lower = message.toLowerCase();
      let responseText = "";

      if (lower.includes("simple words") || lower.includes("simple") || lower.includes("explain")) {
        responseText = `### Simple Breakdown 💡\n\nThink of this like building with LEGO blocks!\n\nWhen we talk about **${message.slice(0, 40)}...**, the main idea is breaking complex tasks into small, self-contained pieces that work together harmoniously.\n\n- **Core takeaway**: Each piece only cares about doing its specific job well.\n- **Real-world analogy**: Like how a car driver only presses the accelerator pedal without needing to manually inject fuel into the engine cylinders!\n\n> 🎯 **Next step**: Would you like a step-by-step example or a quick practice question?`;
      } else if (lower.includes("example") || lower.includes("code")) {
        responseText = `### Concrete Example & Walkthrough 💻\n\nHere is how this works in practice:\n\n\`\`\`javascript\n// Key demonstration of the concept\nfunction executeStudyFlow(subject, difficulty) {\n  console.log(\`Generating optimized lesson for \${subject} [\${difficulty}]\`);\n  return {\n    status: "Mastered",\n    xpEarned: 150,\n    retentionScore: 0.95\n  };\n}\n\nconst session = executeStudyFlow("Data Structures", "Hard");\nconsole.log(session.status);\n\`\`\`\n\n**Why this matters:** Notice how the function keeps logic clean, predictable, and modular.`;
      } else if (lower.includes("exam") || lower.includes("notes") || lower.includes("test")) {
        responseText = `### High-Yield Exam Cheat Sheet 📝\n\nHere are the top points professors love testing on this:\n\n1. **Fundamental Definition**: Ensure you can define the core theorem word-for-word.\n2. **Common Trap**: Watch out for edge cases (e.g., null pointers, zero division, or cycle loops).\n3. **Time/Space Complexity**: Know the average vs worst-case bounds ($O(n \\log n)$ vs $O(n^2)$).\n\n⭐ *Tip*: Spend 5 minutes quizzing yourself on this before sleep to lock in long-term memory!`;
      } else if (lower.includes("quiz") || lower.includes("test me")) {
        responseText = `### Quick Check Quiz 🎯\n\n**Question**: In this concept, what happens when input size doubles in an $O(\\log n)$ operation?\n\n- A) Operations double\n- B) Operations increase by only 1 unit\n- C) Operations quadruple\n- D) Operations remain unchanged\n\n*Reply with your answer and I'll explain why it's right or wrong!*`;
      } else {
        responseText = `### Great question! Let's explore this step-by-step 🚀\n\nRegarding: *"**${message}**"*\n\nHere is the most effective way to understand this:\n\n1. **The Core Principle**: Every system or algorithm is designed to solve an efficiency or organization bottleneck.\n2. **The Mechanism**: It breaks down incoming problems into smaller sub-problems.\n3. **Practical Application**: Used widely in high-performance software, database indexing, and system architecture.\n\n> 💡 **Study Tip**: How would you explain this concept to a friend who has never coded before? Try writing a 1-sentence summary!`;
      }

      return {
        reply: responseText,
        sender: 'ai',
        timestamp: new Date().toISOString()
      };
    }
  },

  // Note Summarization & Key Extraction
  async summarizeNote(noteContent, mode = "summary") {
    try {
      const res = await api.post('/ai/summarize', { content: noteContent, mode });
      return res.data;
    } catch (error) {
      console.warn('Backend offline, using intelligent summarizer fallback', error);
      await new Promise(r => setTimeout(r, 1000));

      const lines = noteContent.split('\n').filter(l => l.trim().length > 0);
      const firstFewLines = lines.slice(0, 3).join(' ').replace(/[#*`]/g, '');

      return {
        summary: `Executive Summary: This study material focuses on ${firstFewLines || "core subject fundamentals"}. Key takeaways emphasize foundational definitions, practical application rules, and critical exam checkpoints.`,
        keyPoints: [
          "Core principles dictate structured data handling and state encapsulation.",
          "Dynamic polymorphism and modular architecture reduce code redundancy.",
          "Attention must be paid to boundary conditions and worst-case execution paths.",
          "High retention is achieved through spaced repetition and self-testing."
        ],
        keyDefinitions: [
          { term: "Primary Concept", definition: "The fundamental theorem or building block governing the subject area." },
          { term: "Invariant Condition", definition: "A property that remains true throughout the entire execution of an algorithm or process." },
          { term: "Efficiency Factor", definition: "Metric measuring resource utilization and time complexity." }
        ],
        examPoints: [
          "Be prepared to compare and contrast alternative implementations under edge conditions.",
          "Memorize standard formulas and time complexity trade-offs.",
          "Review step-by-step problem sets and past paper questions."
        ]
      };
    }
  },

  // Generate Structured AI Quiz
  async generateQuiz({ subject, topic, difficulty = "Medium", questionCount = 5 }) {
    try {
      const res = await api.post('/ai/quiz', { subject, topic, difficulty, questionCount });
      return res.data;
    } catch (error) {
      console.warn('Backend offline, using intelligent quiz generator fallback', error);
      await new Promise(r => setTimeout(r, 1500));

      const generatedQuestions = [];
      const templates = [
        {
          q: `In ${subject} (${topic}), what is the primary purpose of the foundational design pattern?`,
          opts: [
            "To encapsulate state and prevent unintended external mutation",
            "To maximize runtime memory overhead",
            "To disable multithreaded execution",
            "To bypass compile-time syntax verification"
          ],
          correct: 0,
          exp: `Encapsulation and modular design ensure state integrity and prevent bugs caused by unexpected external alterations in ${topic}.`
        },
        {
          q: `Which of the following best describes the worst-case time complexity associated with standard operations in ${topic}?`,
          opts: [
            "O(1) constant time under all circumstances",
            "O(log n) logarithmic reduction",
            "O(n) linear scan or O(n²) under degenerate states",
            "O(n!) factorial exponential explosion"
          ],
          correct: 2,
          exp: `Degenerate conditions (such as skewed trees or unbalanced buckets) can cause performance to degrade from logarithmic to linear or quadratic time.`
        },
        {
          q: `When implementing ${topic} in ${subject}, what is the recommended practice for memory and resource cleanup?`,
          opts: [
            "Ignore unmanaged resources and rely only on eventual termination",
            "Use automated resource management (e.g., try-with-resources or RAII)",
            "Allocate all structures statically in global scope",
            "Duplicate references to avoid garbage collection"
          ],
          correct: 1,
          exp: `Deterministic resource management guarantees that file handles, locks, and network sockets are safely released even if exceptions occur.`
        },
        {
          q: `What key trade-off must be evaluated when choosing between iterative and recursive approaches in ${topic}?`,
          opts: [
            "Code readability vs Call-stack memory overhead",
            "Integer overflow vs floating point precision",
            "Source file length vs compiler licensing",
            "ASCII encoding vs UTF-8 character width"
          ],
          correct: 0,
          exp: `Recursion offers elegant, concise code matching inductive definitions, but risks call stack overflow if depth is unbound. Iteration avoids stack frames.`
        },
        {
          q: `Which diagnostic technique is most effective for isolating boundary bugs in ${topic}?`,
          opts: [
            "Disabling all assertions in production builds",
            "Boundary Value Analysis and testing null/empty/extreme inputs",
            "Randomizing input orders without deterministic seeds",
            "Suppressing compiler warning flags"
          ],
          correct: 1,
          exp: `Boundary Value Analysis systematically checks off-by-one errors, min/max integer bounds, and empty collections where 80%+ of bugs congregate.`
        }
      ];

      const count = Math.min(questionCount, templates.length);
      for (let i = 0; i < count; i++) {
        generatedQuestions.push({
          id: `gen_q_${Date.now()}_${i}`,
          question: templates[i].q,
          options: templates[i].opts,
          correctAnswer: templates[i].correct,
          explanation: templates[i].exp
        });
      }

      return {
        id: `quiz_gen_${Date.now()}`,
        title: `${subject}: ${topic} Mastery Quiz`,
        subject,
        topic,
        difficulty,
        questionCount: count,
        questions: generatedQuestions
      };
    }
  },

  // Generate Flashcards
  async generateFlashcards({ subject, topic, notesContent = "", cardCount = 6 }) {
    try {
      const res = await api.post('/ai/flashcards', { subject, topic, notesContent, cardCount });
      return res.data;
    } catch (error) {
      console.warn('Backend offline, using intelligent flashcards generator fallback', error);
      await new Promise(r => setTimeout(r, 1200));

      const cards = [
        {
          id: `gen_fc_${Date.now()}_1`,
          question: `What is the core definition and role of ${topic || subject}?`,
          answer: `${topic || subject} is a fundamental concept that structures data or logic to ensure optimal performance, modularity, and error resistance.`,
          category: subject || "Core",
          mastered: false
        },
        {
          id: `gen_fc_${Date.now()}_2`,
          question: `What are the 3 critical advantages of applying ${topic || subject}?`,
          answer: "1. High maintainability\n2. Predictable algorithmic scaling\n3. Simplified unit testing and debugging.",
          category: subject || "Core",
          mastered: false
        },
        {
          id: `gen_fc_${Date.now()}_3`,
          question: `What is the standard failure mode or common mistake students make with ${topic || subject}?`,
          answer: "Failing to account for edge conditions (empty inputs, null pointers, race conditions, or unhandled exceptions).",
          category: subject || "Core",
          mastered: false
        },
        {
          id: `gen_fc_${Date.now()}_4`,
          question: `How do you analyze the space-time efficiency of ${topic || subject}?`,
          answer: "Evaluate Big-O bounds for average and worst-case scenarios, considering auxiliary memory allocated during execution.",
          category: subject || "Core",
          mastered: false
        },
        {
          id: `gen_fc_${Date.now()}_5`,
          question: `What key question is frequently asked on midterms regarding ${topic || subject}?`,
          answer: "Comparing its trade-offs against alternative approaches and calculating time-complexity transitions.",
          category: subject || "Core",
          mastered: false
        }
      ];

      return {
        id: `deck_gen_${Date.now()}`,
        title: `${subject}: ${topic} Active Recall Deck`,
        subject: subject || "General",
        cardsCount: cards.length,
        masteredCount: 0,
        cards
      };
    }
  },

  // Generate Intelligent Study Plan
  async generateStudyPlan({ examDate, subjects = [], dailyHours = 3, preferredDays = [] }) {
    try {
      const res = await api.post('/ai/study-plan', { examDate, subjects, dailyHours, preferredDays });
      return res.data;
    } catch (error) {
      console.warn('Backend offline, using intelligent study plan generator fallback', error);
      await new Promise(r => setTimeout(r, 1400));

      const subjectList = subjects.length > 0 ? subjects : ["Computer Science", "Mathematics", "Operating Systems"];
      const days = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"];
      
      const generatedDays = days.map((dayName, idx) => {
        const sub1 = subjectList[idx % subjectList.length];
        const sub2 = subjectList[(idx + 1) % subjectList.length];
        return {
          dayName,
          date: `Day ${idx + 1}`,
          blocks: [
            {
              id: `block_${idx}_1`,
              time: "09:00 - 10:00 AM",
              subject: sub1,
              topic: `${sub1} - Core Concepts & Notes Review`,
              type: "Study",
              duration: "60 mins",
              completed: false
            },
            {
              id: `block_${idx}_2`,
              time: "10:15 - 11:00 AM",
              subject: sub2,
              topic: `${sub2} - Problem Solving & Practice Drills`,
              type: "Practice",
              duration: "45 mins",
              completed: false
            },
            {
              id: `block_${idx}_3`,
              time: "11:00 - 11:20 AM",
              subject: "Break",
              topic: "Hydration, Stretch & Refresh",
              type: "Break",
              duration: "20 mins",
              completed: false
            },
            {
              id: `block_${idx}_4`,
              time: "11:20 - 12:00 PM",
              subject: sub1,
              topic: `${sub1} - Active Recall Flashcards & Mini Quiz`,
              type: "Quiz / Exam Prep",
              duration: "40 mins",
              completed: false
            }
          ]
        };
      });

      return {
        id: `plan_gen_${Date.now()}`,
        title: `AI Personalized Exam Roadmap`,
        examDate: examDate || "Upcoming Exam",
        dailyHours: dailyHours || 3,
        subjects: subjectList,
        targetScore: "95%+",
        createdAt: new Date().toISOString(),
        days: generatedDays
      };
    }
  }
};

File: backend/app/services/ai_service.py
import os
import json
import re
from datetime import datetime
from typing import Dict, Any, List, Optional
from ..core.config import settings

# Attempt to configure Gemini AI if key is present
gemini_available = False
try:
    if settings.GEMINI_API_KEY:
        import google.generativeai as genai
        genai.configure(api_key=settings.GEMINI_API_KEY)
        gemini_available = True
except Exception as e:
    print(f"Gemini AI initialization warning: {e}")

class AIService:
    @staticmethod
    def _clean_json_string(text: str) -> str:
        """Extract valid JSON from markdown code blocks or raw strings."""
        text = text.strip()
        # Remove ```json and ``` wrapping
        match = re.search(r"```(?:json)?\s*([\s\S]*?)\s*```", text)
        if match:
            return match.group(1).strip()
        return text

    @staticmethod
    async def chat(message: str, context: Optional[Dict[str, Any]] = None) -> Dict[str, Any]:
        """Interactive study tutor chat using Gemini or intelligent fallback."""
        tutor_persona = context.get("tutorPersona", "Socratic & Step-by-Step") if context else "Socratic & Step-by-Step"
        
        system_instruction = f"""
You are StudyMate AI, a brilliant, friendly, and patient personal study tutor for college students.
Your role: {tutor_persona}
Guidelines:
1. Explain concepts clearly with simple real-world analogies where helpful.
2. Guide the student toward deep understanding.
3. Use formatted markdown with clear headings, bullet points, and code blocks.
4. Keep responses encouraging, concise, and structured.
"""

        if gemini_available and settings.GEMINI_API_KEY:
            try:
                import google.generativeai as genai
                model = genai.GenerativeModel(
                    model_name="gemini-1.5-flash",
                    system_instruction=system_instruction
                )
                response = model.generate_content(message)
                return {
                    "reply": response.text,
                    "sender": "ai",
                    "timestamp": datetime.utcnow().isoformat()
                }
            except Exception as e:
                print(f"Gemini API error during chat: {e}. Using fallback.")

        # Fallback intelligent tutor response
        lower = message.lower()
        if "simple" in lower or "explain" in lower:
            reply = f"### Simple Concept Breakdown 💡\n\nThink of **{message[:40]}...** like an organized filing cabinet system!\n\n1. **Core Principle**: Breaking down complex structures into modular, reusable components.\n2. **Real-world Analogy**: Just like how a car driver uses standard pedals without needing to manage spark plugs directly.\n\n> 🎯 **Next step**: Would you like a step-by-step example or a quick practice check?"
        elif "example" in lower or "code" in lower:
            reply = f"### Concrete Example 💻\n\nHere is how this is applied in practice:\n\n```python\ndef execute_concept_demo(data):\n    # Process elements efficiently in O(log n)\n    result = [x * 2 for x in data if x > 0]\n    return {'status': 'success', 'processed': len(result)}\n```\n\nNotice how the logic remains clean, predictable, and error-resistant."
        elif "exam" in lower or "cheat" in lower:
            reply = "### High-Yield Exam Watchouts 📝\n\n1. **Definitions**: Ensure you can articulate core theorems without missing keywords.\n2. **Common Trap**: Watch out for edge boundaries (null pointers, division by zero, empty trees).\n3. **Complexity Bounds**: Memorize average vs worst-case trade-offs ($O(n \\log n)$ vs $O(n^2)$)."
        elif "quiz" in lower:
            reply = "### Quick Knowledge Check 🎯\n\n**Question**: When input size doubles in an $O(\\log n)$ operation, how much does computation time increase?\n\n- A) It doubles\n- B) It increases by only 1 constant unit\n- C) It quadruples\n- D) It remains unchanged\n\n*Reply with your answer and I'll explain why it's right or wrong!*"
        else:
            reply = f"### Great question! Let's explore this step-by-step 🚀\n\nRegarding: *\"{message}\"*\n\n1. **Foundational Concept**: It addresses a key architectural and performance bottleneck.\n2. **Mechanism**: It operates by dividing incoming inputs into manageable sub-tasks.\n3. **Practical Application**: Widely utilized in high-performance software and systems.\n\n> 💡 **Study Check**: How would you explain this in one sentence to a classmate?"

        return {
            "reply": reply,
            "sender": "ai",
            "timestamp": datetime.utcnow().isoformat()
        }

    @staticmethod
    async def summarize_note(content: str, mode: str = "all") -> Dict[str, Any]:
        """Summarize study notes and extract structured key takeaways."""
        prompt = f"""
You are an expert academic summarizer. Analyze the following study notes and return a strictly valid JSON object:
{{
  "summary": "2-3 sentence executive summary",
  "keyPoints": ["bullet point 1", "bullet point 2", "bullet point 3", "bullet point 4"],
  "keyDefinitions": [
    {{"term": "Term 1", "definition": "Clear concise definition"}},
    {{"term": "Term 2", "definition": "Clear concise definition"}}
  ],
  "examPoints": ["exam watchout 1", "exam watchout 2"]
}}

Study Notes:
{content}
"""
        if gemini_available and settings.GEMINI_API_KEY:
            try:
                import google.generativeai as genai
                model = genai.GenerativeModel(model_name="gemini-1.5-flash")
                response = model.generate_content(prompt)
                clean_json = AIService._clean_json_string(response.text)
                parsed = json.loads(clean_json)
                return parsed
            except Exception as e:
                print(f"Gemini API error during summarization: {e}. Using fallback.")

        # Fallback structured summary
        first_line = content.split('\n')[0].replace('#', '').strip() if content else "Core Study Topic"
        return {
            "summary": f"Executive Summary: This material provides a structured breakdown of {first_line}. It emphasizes core theoretical definitions, architectural trade-offs, and critical exam checkpoints.",
            "keyPoints": [
                "Fundamental invariants must be preserved throughout state transformations.",
                "Encapsulation and modularity prevent unintended side-effects and bugs.",
                "Worst-case boundary conditions require defensive validation.",
                "Spaced repetition accelerates retention for key formulas and mechanics."
            ],
            "keyDefinitions": [
                {"term": "Primary Principle", "definition": "The foundational theorem governing the subject behavior."},
                {"term": "Invariant", "definition": "A condition that remains true across all valid execution cycles."},
                {"term": "Efficiency Factor", "definition": "Metric measuring computational and memory resource utilization."}
            ],
            "examPoints": [
                "Be ready to compare trade-offs between alternative implementations.",
                "Memorize standard formulas and boundary conditions."
            ]
        }

    @staticmethod
    async def generate_quiz(subject: str, topic: str, difficulty: str = "Medium", question_count: int = 5) -> Dict[str, Any]:
        """Generate structured multiple-choice quiz questions."""
        prompt = f"""
Generate a structured multiple choice quiz with exactly {question_count} questions for college students on:
Subject: {subject}
Topic: {topic}
Difficulty: {difficulty}

Return ONLY a valid JSON object matching this exact schema:
{{
  "title": "{subject}: {topic} Mastery Quiz",
  "subject": "{subject}",
  "topic": "{topic}",
  "difficulty": "{difficulty}",
  "questionCount": {question_count},
  "questions": [
    {{
      "id": "q1",
      "question": "Question text here?",
      "options": ["Option A", "Option B", "Option C", "Option D"],
      "correctAnswer": 0,
      "explanation": "Detailed explanation of why this answer is correct."
    }}
  ]
}}
"""
        if gemini_available and settings.GEMINI_API_KEY:
            try:
                import google.generativeai as genai
                model = genai.GenerativeModel(model_name="gemini-1.5-flash")
                response = model.generate_content(prompt)
                clean_json = AIService._clean_json_string(response.text)
                parsed = json.loads(clean_json)
                return parsed
            except Exception as e:
                print(f"Gemini API error during quiz generation: {e}. Using fallback.")

        # Fallback intelligent structured quiz
        questions = [
            {
                "id": f"q_{i+1}",
                "question": f"In {subject} ({topic}), what is the primary purpose of the core foundational pattern (Concept #{i+1})?",
                "options": [
                    "To encapsulate internal state and maintain modular integrity",
                    "To intentionally introduce quadratic memory overhead",
                    "To disable asynchronous background execution",
                    "To bypass standard runtime validation"
                ],
                "correctAnswer": 0,
                "explanation": f"Encapsulation and modularity protect internal consistency and minimize bugs when handling {topic}."
            }
            for i in range(min(question_count, 5))
        ]

        return {
            "id": f"quiz_gen_{int(datetime.utcnow().timestamp())}",
            "title": f"{subject}: {topic} Mastery Quiz",
            "subject": subject,
            "topic": topic,
            "difficulty": difficulty,
            "questionCount": len(questions),
            "questions": questions
        }

    @staticmethod
    async def generate_flashcards(subject: str, topic: str, notes_content: str = "", card_count: int = 6) -> Dict[str, Any]:
        """Generate structured active-recall flashcard deck."""
        prompt = f"""
Generate an active recall flashcard deck with {card_count} high-yield study cards for:
Subject: {subject}
Topic: {topic}
Extra Notes Content: {notes_content}

Return ONLY a valid JSON object matching this schema:
{{
  "title": "{subject}: {topic} Active Recall Deck",
  "subject": "{subject}",
  "cardsCount": {card_count},
  "masteredCount": 0,
  "cards": [
    {{
      "id": "fc1",
      "question": "Front prompt question?",
      "answer": "Back answer and key explanation.",
      "category": "{subject}",
      "mastered": false
    }}
  ]
}}
"""
        if gemini_available and settings.GEMINI_API_KEY:
            try:
                import google.generativeai as genai
                model = genai.GenerativeModel(model_name="gemini-1.5-flash")
                response = model.generate_content(prompt)
                clean_json = AIService._clean_json_string(response.text)
                parsed = json.loads(clean_json)
                return parsed
            except Exception as e:
                print(f"Gemini API error during flashcard generation: {e}. Using fallback.")

        # Fallback structured flashcards
        cards = [
            {
                "id": f"fc_{i+1}",
                "question": f"What is the key theoretical definition of {topic} (Principle #{i+1})?",
                "answer": f"{topic} structures computational logic and memory layout to ensure deterministic, scalable execution in {subject}.",
                "category": subject,
                "mastered": False
            }
            for i in range(min(card_count, 6))
        ]

        return {
            "id": f"deck_gen_{int(datetime.utcnow().timestamp())}",
            "title": f"{subject}: {topic} Active Recall Deck",
            "subject": subject,
            "cardsCount": len(cards),
            "masteredCount": 0,
            "cards": cards
        }

    @staticmethod
    async def generate_study_plan(exam_date: str, subjects: List[str], daily_hours: float = 3.5, preferred_days: Optional[List[str]] = None) -> Dict[str, Any]:
        """Generate a realistic exam preparation timetable."""
        days_names = ["Monday", "Tuesday", "Wednesday", "Thursday", "Friday", "Saturday"]
        subject_list = subjects if subjects else ["Computer Science", "Operating Systems", "Calculus"]

        generated_days = []
        for idx, day_name in enumerate(days_names):
            s1 = subject_list[idx % len(subject_list)]
            s2 = subject_list[(idx + 1) % len(subject_list)]
            generated_days.append({
                "dayName": day_name,
                "date": f"Day {idx + 1}",
                "blocks": [
                    {
                        "id": f"b_{idx}_1",
                        "time": "09:00 - 10:00 AM",
                        "subject": s1,
                        "topic": f"{s1} - Core Theory & Lecture Notes Review",
                        "type": "Study",
                        "duration": "60 mins",
                        "completed": False
                    },
                    {
                        "id": f"b_{idx}_2",
                        "time": "10:15 - 11:00 AM",
                        "subject": s2,
                        "topic": f"{s2} - Practice Problem Set & Analysis",
                        "type": "Practice",
                        "duration": "45 mins",
                        "completed": False
                    },
                    {
                        "id": f"b_{idx}_3",
                        "time": "11:00 - 11:20 AM",
                        "subject": "Break",
                        "topic": "Hydration, Active Stretch & Walk",
                        "type": "Break",
                        "duration": "20 mins",
                        "completed": False
                    },
                    {
                        "id": f"b_{idx}_4",
                        "time": "11:20 - 12:00 PM",
                        "subject": s1,
                        "topic": f"{s1} - Flashcard Drill & Timed Mini-Quiz",
                        "type": "Quiz / Exam Prep",
                        "duration": "40 mins",
                        "completed": False
                    }
                ]
            })

        return {
            "id": f"plan_gen_{int(datetime.utcnow().timestamp())}",
            "title": "AI Personalized Exam Mastery Plan",
            "examDate": exam_date,
            "dailyHours": daily_hours,
            "subjects": subject_list,
            "targetScore": "95%+",
            "createdAt": datetime.utcnow().isoformat(),
            "days": generated_days
        }
